stroyprombeton/views/catalog.py

```diff
@@ -231,10 +231,38 @@ def partition_(items: list, columns: int) -> typing.Collection[list]:
     )
     return render(
         request,
-        'catalog/series_matrix.html',
+        'catalog/matrix.html',
         {
             'page': page,
-            'parted_series': partition_(series, columns),
+            'parted_items': partition_(series, columns),
+        }
+    )
+
+
+# waiting for #631 to avoid code doubling between series and sections
+def sections_matrix(request, page='sections'):
+    def partition_(items: list, columns: int) -> typing.Collection[list]:
+        """Part items to the columns by given columns count."""
+        result = defaultdict(list)  # Ignore CPDBear
+        column_length = len(items) // columns + 1
+        for i, e in enumerate(items):
+            result[i // column_length].append(e)
+        return result.values()
+
+    columns = settings.SECTIONS_MATRIX_COLUMNS_COUNT
+    page = CustomPage.objects.get(slug=page)
+    sections = (
+        models.Section.objects.bind_fields()
+        .exclude_empty()
+        .order_by('name')
+    )
+    # TODO - solve the problem with Series
```

Comment:
Let's create a properly formatted todo for it or remove

Reply:
sorry, forgot it